### PR TITLE
GPU linear solver interface

### PR DESCRIPTION
**Pull request by pelesh**
This PR creates linear solver interface where ReSolve linear solver can receive data on both CPU and GPU, based on user selection.

### CODE REVIEW

**Comment by pelesh** [comment]
CC @cameronrutherford @ryandanehy

**Review by kswirydo — APPROVED** [review]
I run it on Ascent with exago, it works as intended.

File: src/LinAlg/ReSolve/RefactorizationSolver.cpp [hunk 1]
```diff
@@ -58,7 +58,6 @@
 #include "RefactorizationSolver.hpp"
 
 #include "hiop_blasdefs.hpp"
```
Comment (nychiang) [inline src/LinAlg/ReSolve/RefactorizationSolver.cpp:60]: @pelesh Guess you will remove this line later, once ReSolve is published as a standalone package. Maybe you'd like to address this issue in a later PR.

**Review by nychiang — APPROVED** [review]
LGTM. Just one minor comment.

File: src/LinAlg/ReSolve/RefactorizationSolver.cpp (hunk 1 above)
Comment (pelesh) [reply src/LinAlg/ReSolve/RefactorizationSolver.cpp:60]: Thanks for catching this. I don't think this include is needed at all.

**Review by cnpetra — APPROVED** [review]
Glad this worked out as expected with the cuda backend.

**Pull request merged**